src/routes/posts.js

```diff
@@ -78,9 +78,43 @@ const deletePost = async (req, res, next) => {
     }
 }
 
+// UPDATE AN ARTICLE
+
+const updatePost = async (req, res, next) => {
+    const { id } = req.params // WHERE
+    const { title, content, image, published } = req.body // SET
+
+    try {
+        const result = await db.query(
+            'UPDATE posts SET title = $1, content = $2 , image = $3 , published = $4 WHERE id = $5 RETURNING *',
+            [title, content, image, published, id]
+        )
+        const updatedArticle = result.rows[0]
```

Comment:
i think the error  handling you're doing now is cool but for this particular usecase, it will be good for whoever is calling the endpoint to know that the post did not exist hence the error. The current generic one doesn't say much

So lets instead check if the `result.rows[0]` does not exist and then return even a `404 Not Found` or maybe more appropriate codes that post does not exist before returning the success response

something like
```js
if(!result.rows[0]){
  return res.status(404).json({success: false, message:: 'article so so and so'})
}
```

Reply:
thank you, how about the new changes i just pushed?